dask/dataframe/shuffle.py

```diff
@@ -676,8 +707,9 @@ def compute_divisions(df, **kwargs):
             "Partitions must be sorted ascending with the index", mins, maxes
         )
 
+    flag = False
     if any(a >= b for a, b in zip(maxes[:1], mins[1:])):
-        warnings.warn("Partition indices have overlap.")
+        flag = True
 
     divisions = tuple(mins) + (list(maxes)[-1],)
```

Comment:
I wonder if we might call `fix_overlap` here rather than return a flag and then call it.  This might reduce some of the logic.

Also, if we wanted to be clever we could fix overlap only on those partitions that need it.  We have a bit more detailed information at this point with mins/maxes than we do once we leave this function. This is probably a premature optimization though.